### ISSUE TITLE
Estimators created from Keras models expect unconsumed outputs to be provided with data in training/evaluation.

### ISSUE THREAD

**Issue by zmjjmz** [tensorflow/tensorflow#21811]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: Yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Debian Jessie
- **TensorFlow installed from (source or binary)**: Binary
- **TensorFlow version (use command below)**: 1.9
- **Python version**: 2.7.9
- **Exact command to reproduce**: See gist [here](https://gist.github.com/zmjjmz/3f621aaafc5683238ade6224e5dedcb6)
- **Bazel version**: N/A
- **CUDA/cuDNN version**: N/A
- **GPU model and memory**: N/A
- **Mobile device**: N/A

### Describe the problem
Sorry, this is a bit arcane/edge-casey. Basically I have a Keras model that I want to convert into an Estimator for use with the `train_and_evaluate` framework. One aspect of these models is that I have layers that are outputs but do not contribute to the loss that are part of the deployed model's contract -- e.g. an `oov_code` that indicates if the input (text) was out-of-vocabulary. 

The way I've been doing this _before_ switching to a `keras.model_to_estimator` method was to have the Keras model's loss and metrics be set to just the parts contribute to the loss in the `ModeKeys.TRAIN` or `ModeKeys.EVAL` modes and then have the `export_outputs` argument take all of them in the `PREDICT` mode, e.g. in this [gist](https://gist.github.com/zmjjmz/b84248a8a1ff21afd1c172e0abbba2e8). I can't do this fine-grained thing in `model_to_estimator`, which is sort of the first problem.

However, theoretically the `Model.compile` function allows for the `loss` and `metrics` arguments to be dictionaries of output names (derived from the layer names) to their respective loss / metric functions. When using `Model.fit`, one can (as in line 49 of the provided reproduction) provide a dictionary of `output:labels` which are then appropriately routed to their respective losses / metrics (as can be seen by running it - note the `classes_accuracy` and `classes_loss` printouts.

Unfortunately, when using `model_to_estimator` followed by `train_and_evaluate` my eternal enemy `_create_ordered_io` expects to find a value in `y` for all outputs, irrespective of the `loss` and `metrics` not specifying any such requirements. 

Confusingly I am promised separately that `we will not be expecting any data to be passed to "reporting" during training.` as shown in the provided logs.

Basically, it would be nice if `_create_ordered_io` had some way to 'know' that these outputs should be ignored for these steps, OR for a way to specify different outputs according to the `ModeKeys` in `model_to_estimator`. I can workaround this at the moment by providing a fake output in the `numpy_input_fn`'s `y` argument (e.g. what happens if you set `incl_reporting=True` in the `get_input_fn` function in the reproduction script), however this is really not ideal for a variety of reasons.

### Source code / logs
The gist provided [here](https://gist.github.com/zmjjmz/3f621aaafc5683238ade6224e5dedcb6) has both reproduction code and comments with the outputs.

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Bazel version
CUDA/cuDNN version
GPU model and memory
Mobile device

**Comment by zmjjmz** [comment]
Seems related, but I guess the issue is that while the Keras APIs don't expect that data the Estimator ones do.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @fchollet: It has been 20 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @fchollet: It has been 35 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @fchollet: It has been 50 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @fchollet: It has been 65 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @fchollet: It has been 80 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @fchollet: It has been 95 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by ymodak** [comment]
This is fixed with latest tf-nightly version '1.15.0-dev20190726'. Thanks!
